Tools/scripts/build_binaries.py

```diff
@@ -685,6 +689,103 @@ def buildlogs_dirpath(self):
         return os.getenv("BUILDLOGS",
                          os.path.join(os.getcwd(), "..", "buildlogs"))
 
+    def create_pdef_xml_file(self, vehicle_type: str, dst_dir: str, git_tag: str) -> bool:
+        '''generate parameter metadata XML file for a specific vehicle and version'''
+        self.progress(f"Generating apm.pdef.xml for {vehicle_type} {git_tag}")
+
+        try:
+            # Run param_parse.py to generate metadata
+            # Note: We're already at the correct git tag from the build process
+            param_parse_path = os.path.join(topdir(), 'Tools', 'autotest',
+                                            'param_metadata', 'param_parse.py')
+
+            if not os.path.exists(param_parse_path):
+                self.progress(f"param_parse.py not found at {param_parse_path}")
+                return False
+
+            self.run_program(
+                'PARAM-PARSE',
+                ['python3', param_parse_path, '--vehicle', vehicle_type, '--format', 'xml'],
+                show_output=True
+            )
+
+            # Check if the output file was created
+            apm_pdef_path = os.path.join(topdir(), 'apm.pdef.xml')
```

Comment:
Should we remove this file before we attempt to generate it?  Might stop us using a stale file inadvertently.